<!-- Launch file for the rrt-detector and the assigner -->


<launch>
    <arg name="eta" value="1.0"/>
    <arg name="Geta" value="1.0"/>
    <arg name="info_radius" value="1.0"/>
    <arg name="info_multiplier" value="3.0"/>
    <arg name="hysteresis_radius" value="3.0"/>
    <arg name="hysteresis_gain" value="2.0"/>
    <arg name="costmap_clearing_threshold" value="50"/>
    <arg name="goal_time_thres" value="45"/>
    <arg name="delay_after_assignement" value="3.8"/>   
    <arg name="frequency" value="50"/>
    <!-- robot name space definition -->
    <arg name="first_tb3"  default="tb3_1"/>
    <arg name="second_tb3" default="tb3_2"/>
    <arg name="third_tb3"  default="tb3_0"/>
  
  
    <node pkg="rrt_exploration" type="global_rrt_detector" name="global_rrt_detector" output="screen">
      <param name="eta" value="$(arg Geta)"/>
      <param name="map_topic" value="/map"/>
    </node>
  
    <node pkg="rrt_exploration" type="local_rrt_detector" name="$(arg first_tb3)_rrt_detector" output="screen">
      <param name="eta" value="$(arg eta)"/>
      <param name="map_topic" value="/$(arg first_tb3)/map"/>
      <param name="robot_frame" value="/$(arg first_tb3)/base_link"/>
    </node>
  
    <node pkg="rrt_exploration" type="local_rrt_detector" name="$(arg second_tb3)_rrt_detector" output="screen">
      <param name="eta" value="$(arg eta)"/>
      <param name="map_topic" value="/$(arg second_tb3)/map"/>
      <param name="robot_frame" value="/$(arg second_tb3)/base_link"/>
    </node>
  
    <node pkg="rrt_exploration" type="local_rrt_detector" name="$(arg third_tb3)_rrt_detector" output="screen">
      <param name="eta" value="$(arg eta)"/>
      <param name="map_topic" value="/$(arg third_tb3)/map"/>
      <param name="robot_frame" value="/$(arg third_tb3)/base_link"/>
    </node>
    
  
    <node pkg="rrt_exploration" type="filter.py" name="filter" output="screen">
      <param name="map_topic" value="/map"/>
      <param name="namespace_init_count" value="0"/>
      <param name="info_radius" value="$(arg info_radius)"/> 
      <param name="costmap_clearing_threshold" value="$(arg costmap_clearing_threshold)"/> 
      <param name="goals_topic" value="/detected_points"/> 
      <param name="n_robots" value="3"/>
      <param name="namespace" value="/tb3_"/>
      <param name="rate" value="$(arg frequency)"/>
    </node>
    
    
    <node pkg="rrt_exploration" type="assigner.py" name="assigner" output="screen">
      <param name="map_topic" value="/map"/>
      <param name="global_frame" value="map"/>
      <param name="namespace_init_count" value="0"/>
      <param name="info_radius" value="$(arg info_radius)"/> 
      <param name="info_multiplier" value="$(arg info_multiplier)"/> 
      <param name="hysteresis_radius" value="$(arg hysteresis_radius)"/> 
      <param name="hysteresis_gain" value="$(arg hysteresis_gain)"/> 
      <param name="frontiers_topic" value="/filtered_points"/> 
      <param name="n_robots" value="3"/>
      <param name="namespace" value="/tb3_"/>
      <param name="goal_time_thres" value="$(arg goal_time_thres)"/>
      <param name="delay_after_assignement" value="$(arg delay_after_assignement)"/>
      <param name="rate" value="$(arg frequency)"/>
    </node>
  </launch>
